Gems/Atom/RPI/Code/Source/RPI.Public/Pass/AttachmentsReadbackGroup.cpp

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RPI.Public/Buffer/BufferSystemInterface.h>
+#include <Atom/RPI.Public/Buffer/Buffer.h>
+#include <Atom/RPI.Public/Image/AttachmentImage.h>
+#include <Atom/RPI.Public/Image/ImageSystemInterface.h>
+#include <Atom/RPI.Public/Pass/AttachmentsReadbackGroup.h>
+#include <Atom/RPI.Public/RPIUtils.h>
+
+#include <Atom/RPI.Reflect/Buffer/BufferAssetCreator.h>
+
+#include <Atom/RHI/CommandList.h>
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/Fence.h>
+#include <Atom/RHI/FrameGraphExecuteContext.h>
+#include <Atom/RHI/FrameScheduler.h>
+#include <Atom/RHI/RHISystemInterface.h>
+#include <Atom/RHI/RHIUtils.h>
+#include <Atom/RHI/ScopeProducerFunction.h>
+
+#include <AzCore/Serialization/Json/JsonUtils.h>
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+namespace AZ
+{
+    namespace RPI
```

Comment:
very minor but we can save an indent level by using
```suggestion
namespace AZ::RPI
{
```

Reply:
Thanks for the feedback. I'm also waiting for @VickyAtAZ feedback if she has a preference to integrate the new functionality into the existing AttachmentReadback class or keep this original design that introduces a new subclass called AttachmentsReadbackGroup. 

The more I look at the code, the more "clunky" AttachmentsReadbackGroup feels to me, but if no one else feels the clunkiness, I'll leave it as is.

Reply:
@AMZN-alexpete , after talking to Vicky, she also agreed that it was better to add the new capabilities to `AttachmentReadback`. So I've made the corresponding changes and removed `AttachmentsReadbackGroup`. Decided not to do the `namespace AZ::RPI` suggestion because it would impact ALL the file and make code review harder.